lib/providers/power_trait_provider.dart

```diff
@@ -7,13 +8,11 @@ typedef GetDeviceDetailsMethod = Future<Device> Function(
 typedef SendPowerMethod = Future<void> Function(
     Request request, String id, bool onOff);
 
-enum PowerState { idle, loading, performingAction, error }
```

Comment:
+1 on this. We can probably generalize this State enum to all "Providers" (ChangeNotifiers)

Reply:
Agreed. We also can probably factor out the state as a whole in an abstract provider but I was getting weary of touching too much right now.